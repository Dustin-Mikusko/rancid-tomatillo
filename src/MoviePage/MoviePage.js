import React from 'react';
import RatingForm from '../Containers/RatingForm/RatingForm';
import PropTypes from 'prop-types';
import './MoviePage.scss'

const MoviePage = ({ id, title, release_date, poster_path, backdrop_path, overview, average_rating, user }) => {
  let userRating;
  if (user && user.ratings.find(rating => rating.movie_id === id)) {
    userRating = user.ratings.find(rating => rating.movie_id === id)
  }

  let releaseDate = release_date.split('-').reverse().join('/').slice(3);

  let avgRating = average_rating.toFixed();
  

  return (
    <section className='movie-page'>
      <img src={backdrop_path} className='backdrop'/>
      <div className='movie-page-info'>
        <h1 className='movie-page-title'>{title}</h1>
        <p className='release-date'>Released: {releaseDate}</p>
        <p className='overview'>{overview}</p>
        <div className='page-ratings'>
          <p>Avg Rating: {avgRating}</p>
          <p className='your-rating'>{userRating ? `Your Rating: ${userRating.rating}` : ''}</p>
        </div>
          {user && <RatingForm userId={user.id} ratingId={userRating ? userRating.id : undefined} movieId={id}/>}
      </div>
    </section>
  )
}

export default MoviePage;

MoviePage.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  release_date: PropTypes.string,
  poster_path: PropTypes.string,
  backdrop_path: PropTypes.string,
  overview: PropTypes.string,
  average_rating: PropTypes.number,
  user: PropTypes.object
}
