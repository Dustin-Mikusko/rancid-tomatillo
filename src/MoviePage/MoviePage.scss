* {
  box-sizing: border-box;
  color: #FFF;
}

.movie-page {
  height: 75vh;
  margin: auto;
  width: 100%;
}

.backdrop {
  filter: brightness(.5);
  position: relative;
  width: 85%;
  z-index: -2;
}

.movie-page-info {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  margin-left: 25%;
  margin-top: -510px;
  width: 50%;
}

.movie-page-title {
  font-size: 350%;
}

.release-date {
  font-size: 125%;
  font-weight: 700;
}

.overview {
  font-size: 112%;
  font-weight: 700;
}

.page-ratings {
  display: flex;
  flex-direction: row;
  font-size: 125%;
  font-weight: 700;
  justify-content: space-between;
}