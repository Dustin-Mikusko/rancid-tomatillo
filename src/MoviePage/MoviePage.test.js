import React from 'react';
import MoviePage from './MoviePage';
import { shallow } from 'enzyme';

describe('MoviePage', () => {
  it('should match the snapshot when there is no user', () => {
    const wrapper = shallow(<MoviePage />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is a user and a userRating', () => {
    const mockUser = {
      ratings: [
        { name: 'Knives Out', movie_id: 10 },
        { name: 'Moana', movie_id: 7 }
      ]
    };

    const mockId =7;

    const wrapper = shallow(<MoviePage user={mockUser} id={mockId} />);

    expect(wrapper).toMatchSnapshot();
  })
})
