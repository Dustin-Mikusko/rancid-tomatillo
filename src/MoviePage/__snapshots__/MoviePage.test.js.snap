// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match the snapshot when there is a user and a userRating 1`] = `ShallowWrapper {}`;

exports[`MoviePage should match the snapshot when there is no user 1`] = `ShallowWrapper {}`;
